### PR TITLE
Adding ASRHeathError object type change message, E2A Fix for TFO with Recovery Point , A2A Replication Group fix

### PR DESCRIPTION
**Pull request by viverm**
****

## Description

Added deprecated message for object type change .

Bug fixes 
     ASR E2A Fix for TFO recovery Plan  
     ASR A2A Replication Group fix

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.SiteRecovery/Fabrics/StartAzureRmRecoveryServicesAsrSwitchProcessServerJob.cs [hunk 1]
```diff
@@ -24,6 +25,7 @@ namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
     [Cmdlet("Start", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesAsrSwitchProcessServerJob",DefaultParameterSetName = ASRParameterSets.Default,SupportsShouldProcess = true)]
     [Alias("Start-ASRSwitchProcessServerJob")]
     [OutputType(typeof(ASRJob))]
+    [GenericBreakingChange("Property type ASRHealthError_2016_08_10 changed to ASRHealthError")]
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/Fabrics/StartAzureRmRecoveryServicesAsrSwitchProcessServerJob.cs]: Why is this necessary?  If this is just a change in namespace, then there is no need for a breaking change entry.  If there are changed properties, you need to describe them more thoroughly.  This applies to all similar changes below.

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs [hunk 2]
```diff
@@ -611,6 +613,11 @@ private void AzureToAzureReplication(EnableProtectionInput input)
                 RecoveryBootDiagStorageAccountId = this.RecoveryBootDiagStorageAccountId
             };
 
+            if (!string.IsNullOrEmpty(this.ReplicationGroupName))
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs]: Use PSCmdlet.IsParameterBound pattern, and protect this parameter with a ValidateNotNullOrEmpty attribute

File: src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs [hunk 3]
```diff
@@ -56,6 +57,8 @@ public class SetAzureRmRecoveryServicesAsrReplicationProtectedItem : SiteRecover
         ///     Gets or sets the NIC of the virtual machine for which this cmdlet sets the recovery network property.
         /// </summary>
         [Parameter]
+        [CmdletParameterBreakingChange("PrimaryNic", ChangeDescription = "Further this property will be used to set primary NIC for Replication ProtectedItem."
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs]: You need to describe what the actual change is - how will a user need to update their scripts for thsi change?

Comment (viverm) [reply src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs]: [CmdletParameterBreakingChange("PrimaryNic", ChangeDescription = "Further this property will be used to set primary NIC for Replication ProtectedItem."
            + "And UpdateNic will be used to update(selected) provided NIC",
            OldWay = "Set-AzRecoveryServicesAsrReplicationProtectedItem -ReplicationProtectedItem $RPI -PrimaryNic $NicId -RecoveryNetworkId $AzureNetworkID -RecoveryNicSubnetName $subnetName",
            NewWay = "[To update the details for selected NIC] Set-AzRecoveryServicesAsrReplicationProtectedItem -InputObject $RPI -UpdateNic $NicId -RecoveryNicSubnetName $subnetName"
            + "[To update the primary NIC to be used after failover] Set-AzRecoveryServicesAsrReplicationProtectedItem -InputObject $RPI -PrimaryNic $NicId")]

Is there the way to add line change in "NewWay"

**Comment by viverm** [comment]
@markcowl  are changes fine,

File: src/RecoveryServices/RecoveryServices/help/Register-AzRecoveryServicesBackupContainer.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ schema: 2.0.0
 # Register-AzRecoveryServicesBackupContainer
 
 ## SYNOPSIS
-This command allows Azure Backup to convert the �Resource� to a �Backup Container� which is then registered to the given Recovery services vault. The Azure Backup service can then discover workloads of the given workload type within this container to be protected later.
+This command allows Azure Backup to convert the �Resource� to a �Backup Container� which is then registered to the given Recovery services vault. The Azure Backup service can then discover workloads of the given workload type within this container to be protected later.
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices/help/Register-AzRecoveryServicesBackupContainer.md]: Need to fix the encoding of this file

File: src/RecoveryServices/RecoveryServices/help/Register-AzRecoveryServicesBackupContainer.md [hunk 5]
```diff
@@ -33,7 +33,7 @@ The cmdlet registers an Azure VM for AzureWorkloads with specific workloadType.
 
 ### Example 1
 ```
-PS C:\> Register-AzRecoveryServicesBackupContainer -ResourceId <AzureVMID> -VaultId <vaultID> -WorkloadType �MSSQL� -BackupManagementType �AzureWorkload�
+PS C:\> Register-AzRecoveryServicesBackupContainer -ResourceId <AzureVMID> -VaultId <vaultID> -WorkloadType �MSSQL� -BackupManagementType �AzureWorkload�
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices/help/Register-AzRecoveryServicesBackupContainer.md]: same here

File: src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupContainer.md [hunk 6]
```diff
@@ -78,7 +78,7 @@ The acceptable values for this parameter are:
 Type: Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets.Models.ContainerType
 Parameter Sets: (All)
 Aliases:
-Accepted values: AzureVM, Windows, AzureSQL, AzureStorage, AzureWorkload
+Accepted values: AzureVM, Windows, AzureSQL, AzureStorage, AzureVMAppContainer
```
Comment (markcowl) [inline src/RecoveryServices/RecoveryServices/help/Get-AzRecoveryServicesBackupContainer.md]: This looks like a breaing change.  Is the `AzureWorkload` value no longer supported?

**Comment by cormacpayne** [comment]
@viverm ping on the comments above

**Comment by cormacpayne** [comment]
Closing this PR due to lack of activity. Feel free to re-open the PR whenever you're able to address the above comments.
